From 825478ee27f916553938afaf5164fec22cb32732 Mon Sep 17 00:00:00 2001
From: Johannes Meixner <jsmeix@suse.com>
Date: Tue, 10 May 2022 10:43:39 +0200
Subject: [PATCH] Update 200_partition_layout.sh

In layout/save/GNU/Linux/200_partition_layout.sh
ensure $disk_label is one of the supported partition tables, cf.
https://github.com/rear/rear/issues/2801#issuecomment-1122015129
---
 usr/share/rear/layout/save/GNU/Linux/200_partition_layout.sh | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/usr/share/rear/layout/save/GNU/Linux/200_partition_layout.sh b/usr/share/rear/layout/save/GNU/Linux/200_partition_layout.sh
index b747d121c6..6e641d280e 100644
--- a/usr/share/rear/layout/save/GNU/Linux/200_partition_layout.sh
+++ b/usr/share/rear/layout/save/GNU/Linux/200_partition_layout.sh
@@ -115,6 +115,11 @@ extract_partitions() {
         parted -s $device print > $TMP_DIR/parted
         disk_label=$(grep -E "Partition Table|Disk label" $TMP_DIR/parted | cut -d ":" -f "2" | tr -d " ")
     fi
+    # Ensure $disk_label is valid to determine the partition name/type in the next step at 'declare type'
+    # cf. https://github.com/rear/rear/issues/2801#issuecomment-1122015129
+    if ! [[ "$disk_label" = "msdos" || "$disk_label" = "gpt" || "$disk_label" = "gpt_sync_mbr" || "$disk_label" = "dasd" ]] ; then
+        Error "Unsupported partition table '$disk_label' (must be one of 'msdos' 'gpt' 'gpt_sync_mbr' 'dasd')"
+    fi
 
 
     cp $TMP_DIR/partitions $TMP_DIR/partitions-data

